### ISSUE TITLE
Enable feature (e.g., highlighting) for a particular filetype(s)

### ISSUE THREAD

**Issue by andymass** [nvim-treesitter/nvim-treesitter#991]
**Is your feature request related to a problem? Please describe.**

I'd like to enable highlighting for a select few filetypes (for example cpp).  There seems to be no way to do this.  I tried for example (which evidently isn't supported),

```
  highlight = {
    enable = { "cpp" },
  },
```

I also tried
```
  highlight = {
    enable = false,
  },
```

Then `TSBufEnable highlight`, but that also didn't work.

**Describe the solution you'd like**

Allow either of the above approaches (option naming is not critical) 

**Describe alternatives you've considered**

Could maybe do this through autocmds + lua by calling nvim-treesitter's internal apis.

**Additional context**

Users should be able to "opt in" not just "opt out" of features on a per-filetype basis.

**Comment by kyazdani42** [comment]
this could be a good time to rework the internals of the setup which is a bit clunky imo.

**Comment by justuswilhelm** [comment]
I agree, whitelisting would be really helpful for when you want to try nvim-treesitter on only one or two languages for the beginning.

**Comment by clason** [comment]
see #4767 

TL;DR: with 1.0 (WIP on the `main` branch), features can (and have to) be enabled individually in a `FileType` autocommand or `ftplugin`. Nvim-treesitter will only handle parser and query installation and management.
